internal/pkg/s3/s3.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package s3 provides a client to retrieve Copilot S3 information.
+package s3
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/copilot-cli/internal/pkg/aws/resourcegroups"
+	"github.com/aws/copilot-cli/internal/pkg/aws/s3"
+	"github.com/aws/copilot-cli/internal/pkg/deploy"
+)
+
+const (
+	bucketType = "s3:bucket"
+)
+
+type resourceGetter interface {
+	GetResourcesByTags(resourceType string, tags map[string]string) ([]*resourcegroups.Resource, error)
+}
+
+// Client retrieves Copilot S3 service information from AWS.
+type Client struct {
+	rgGetter resourceGetter
+}
+
+// New inits a new Client.
+func New(sess *session.Session) *Client {
+	return &Client{
+		rgGetter: resourcegroups.New(sess),
+	}
+}
+
+// BucketName returns the bucket name given the Copilot app, env, and Static Site service name.
```

Comment:
Will this return buckets created in addons stacks? They should be tagged with the same app, env, svc tags as the main services. 

Can we add some more detail to the documentation describing this, if true, or add some error checking logic on svcType or bucketName? Or perhaps this validation is better done a level higher in the abstraction.

Reply:
Maybe we keep the logic for querying the real name for ECS service and S3 bucket as is for now? We can always replace the logic with checking CFN stack to find out instead of using tags in the future (until they have a valid use case)